core/GlobalState.cc

```diff
@@ -98,6 +103,16 @@ GlobalState::GlobalState(shared_ptr<ErrorQueue> errorQueue, shared_ptr<lsp::Type
     ENFORCE((namesByHashSize & (namesByHashSize - 1)) == 0, "namesByHashSize is not a power of 2");
 }
 
+unique_ptr<GlobalState> GlobalState::makeEmptyGlobalStateForHashing(spdlog::logger &logger) {
+    // Note: Private constructor.
+    unique_ptr<GlobalState> rv(
+        new GlobalState(make_shared<core::ErrorQueue>(logger, logger, make_shared<core::NullFlusher>()),
+                        make_shared<lsp::TypecheckEpochManager>(), 999));
```

Comment:
WDYT about making this `INT_MIN` or something that's not likely to be a "real" id?  (-1 is used elsewhere for other purposes.)  IIUC, the id isn't even mixed in with the hash, so the value shouldn't matter much...but an unusual negative number might stand out a bit more in a debugging session.

Reply:
Works for me 👍